src/serializers/type_serializers/union.rs

```diff
@@ -198,24 +202,180 @@ impl TypeSerializer for UnionSerializer {
     }
 }
 
-pub struct TaggedUnionBuilder;
+#[derive(Debug, Clone)]
+pub struct TaggedUnionSerializer {
+    discriminator: Discriminator,
+    lookup: HashMap<String, CombinedSerializer>,
+    choices: Vec<CombinedSerializer>,
+    name: String,
+}
 
-impl BuildSerializer for TaggedUnionBuilder {
+impl BuildSerializer for TaggedUnionSerializer {
     const EXPECTED_TYPE: &'static str = "tagged-union";
 
     fn build(
         schema: &Bound<'_, PyDict>,
         config: Option<&Bound<'_, PyDict>>,
         definitions: &mut DefinitionsBuilder<CombinedSerializer>,
     ) -> PyResult<CombinedSerializer> {
-        let schema_choices: Bound<'_, PyDict> = schema.get_as_req(intern!(schema.py(), "choices"))?;
-        let mut choices: Vec<CombinedSerializer> = Vec::with_capacity(schema_choices.len());
+        let py = schema.py();
+        let discriminator = Discriminator::new(py, &schema.get_as_req(intern!(py, "discriminator"))?)?;
+
+        let choice_list: Bound<PyDict> = schema.get_as_req(intern!(py, "choices"))?;
```

Comment:
IMO `choice_list` is an odd name for a `dict`, maybe just:

```suggestion
        let choices: Bound<PyDict> = schema.get_as_req(intern!(py, "choices"))?;
```

Reply:
Haha, it certainly is. Will do!